tests/test_aggregates.py

```diff
@@ -1,362 +1,375 @@
+        # this also tests that the aggregator functions (groupsof, groupby) are
+        # generating tuples internally.
+        aggregate_store = self.create_aggregate_store(aggregator_instance, project)
+        if not isinstance(aggregate_store, _DefaultAggregateStore):
+            for aggregate in aggregate_store._generate_aggregates():
+                assert isinstance(aggregate, tuple)
+                assert all(isinstance(job, signac.contrib.job.Job) for job in aggregate)
+        for aggregate in aggregate_store.values():
+            assert isinstance(aggregate, tuple)
+            assert all(isinstance(job, signac.contrib.job.Job) for job in aggregate)
+
+    @pytest.mark.parametrize(
+        "aggregator_instance", AggregateFixtures.list_of_aggregators()
+    )
+    def test_get_by_id(self, project, aggregator_instance):
+        # Ensure that all aggregates can be fetched by id.
+        aggregate_store = self.create_aggregate_store(aggregator_instance, project)
+        for aggregate in aggregate_store.values():
+            assert aggregate == aggregate_store[get_aggregate_id(aggregate)]
+
+    def test_get_invalid_id(self, project):
+        jobs = tuple(project)
+        full_aggregate_store = self.create_aggregate_store(aggregator(), project)
+        default_aggregate_store = self.create_aggregate_store(
+            aggregator.groupsof(1), project
+        )
+        # Test for an aggregate of single job for aggregator instance
+        with pytest.raises(LookupError):
+            full_aggregate_store[get_aggregate_id((jobs[0],))]
+        # Test for an aggregate of all jobs for default aggregator
+        with pytest.raises(LookupError):
+            default_aggregate_store[get_aggregate_id(jobs)]
+
+    def test_contains(self, project):
+        jobs = tuple(project)
+        full_aggregate_store = self.create_aggregate_store(aggregator(), project)
+        default_aggregate_store = aggregator.groupsof(1)._create_AggregateStore(project)
+        # Test for an aggregate of all jobs
+        assert get_aggregate_id(jobs) in full_aggregate_store
+        assert get_aggregate_id(jobs) not in default_aggregate_store
+        # Test for an aggregate of single job
+        assert not jobs[0].id in full_aggregate_store
+        assert jobs[0].id in default_aggregate_store
+
+
+# Test the decorator class aggregator
+class TestAggregate(AggregateProjectSetup, AggregateFixtures):
+    @pytest.mark.parametrize("agg_value", [(1, 2, 3, 4, 5), (), [1, 2, 3, 4, 5], []])
+    def test_default_init(self, agg_value):
+        aggregate_instance = aggregator()
+        assert not aggregate_instance._is_default_aggregator
+        assert aggregate_instance._sort_by is None
+        assert aggregate_instance._sort_ascending
+        assert aggregate_instance._select is None
+        assert list(aggregate_instance._aggregator_function(agg_value)) == [
+            tuple(agg_value)
+        ]

```

Comment:
```suggestion
        assert next(aggregate_instance._aggregator_function(agg_value)) == tuple(agg_value)
```
Also, this check seems a bit off to me. I get that we just wrap anything passed in a tuple by default, but this isn't really what the default aggregator is for. At the very least we should comment why this check exists.

Reply:
I don't think we should be choosing the first element here using `next` because we need to make sure that the output of `_aggregator_function` is everything grouped into 1.

Reply:
I get your thoughts, just testing the aggregator on random input feels off. We also don't need to parameterize this test.

Reply:
Hmm, let's remove the parameters but are you suggesting to remove this test instead?

Reply:
I removed the parameters, let me know how it looks